### PR TITLE
Feature/tfs task refs

### PR DESCRIPTION
**Pull request by keyvanm**
I think I accidentally merged in the RTC task references branch too. If that causes problems please let me know so I can clean it up.

### CODE REVIEW

**Comment by keyvanm** [comment]
if I change the title of a task in TFS (or Jira) and sync again, the old task gets recreated and then I get this error:
`APICallError: HTTP Error 400. Explanation returned: FATAL ERROR: {"errors":["Task reference with this Task and Alm connection already exists."]}`

Error happens in TFS and Jira (the ones I tested).

**Comment by sibartlett** [comment]
The HTTP 400 error you described, is definitely a bug.

sdetools should be using the task reference information to load the JIRA or TFS task - even if their title has been changed.

File: sdetools/modules/sync_tfs/tfs_plugin.py [hunk 1]
```diff
@@ -163,8 +179,9 @@ def alm_validate_configurations(self):
             raise AlmException('Invalid field(s) specified: {0}'.format(', '.join(invalid_fields)))
 
     def alm_get_task(self, task):
```
Comment (sibartlett) [inline sdetools/modules/sync_tfs/tfs_plugin.py]: `alm_get_task` should be renamed to `alm_get_task_legacy`.

It also should not call `alm_get_task_by_reference`, you can remove the first 3 lines of the method.

The base class' `alm_get_task` method determines which of the 2 methods to call.

**Comment by MartinPetkov** [comment]
@keyvanm @sibartlett I could not reproduce the HTTP 400 bug, as it was explained. These are the steps I took:
1. Ran the initial sync between SDE and Jira
2. Went on Jira and changed the title of one of the tasks that was synced
3. Ran the sync again

I did not encounter any bugs.

**Comment by sibartlett** [comment]
I've cleaned this branch up, and opened a new merge request here: #412

**Pull request closed without merging**